src/lib/Manipulator.tsx

```diff
@@ -1,21 +1,28 @@
-import { CSSProperties, useState } from "react";
+import { CSSProperties, useState, useCallback } from "react";
 import { useDragAndZoom } from "./useDragAndZoom";
 import { useBoundsContext } from "./BoundsManager";
+import {Bounds} from "./basic-types";
 
 interface ManipulatorProps {
     style: CSSProperties;
     onHover?: (x: number, event: PointerEvent) => void;
     onHoverEnd?: () => void;
     onTouchUp?: (x: number, event: PointerEvent) => void;
+    onChangeBounds?: (bounds: Bounds) => void;
 }
 
 export function Manipulator(props: ManipulatorProps) {
-    const { onHover, onHoverEnd, onTouchUp } = props;
+    const { onHover, onHoverEnd, onTouchUp, onChangeBounds } = props;
     const { xBoundsLimit, xBoundsMinVisible, settledXBounds, onManipulation, onManipulationEnd } = useBoundsContext();
 
     const [glass, setGlass] = useState<HTMLDivElement | null>(null);
 
-    useDragAndZoom(glass, settledXBounds, onManipulation, onManipulationEnd, onHover, onHoverEnd, onTouchUp, {
+    const onEnd = useCallback((bounds: Bounds) => {
+        onManipulationEnd(bounds);
+        onChangeBounds?.(bounds);
+    }, [onManipulationEnd])
```

Comment:
`onEnd` calls `onChangeBounds`, but doesn't have it in dependency array. This may cause calling an obsolete copy of the callback.

But instead of putting `onChangeBounds` into the deps array, I suggest to use `useRef` or `useLatest` like this. This will prevent `useDragAndZoom` from rebinding every time `onChangeBounds` callback changes.

```typescript
const { onChangeBounds } = props;

const latestChangeBounds = useLatest(onChangeBounds);

const onEnd = useCallback(() => {
   ...
   latestChangeBounds.current?.(bounds);
}, [...]);
```